spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+describe Api::V1::UsersController, type: :controller do
+
+  let (:link1) { create(:link) }
+  let (:link2) { create(:link) }
+  let(:user) { create(:user) }
+
+  describe "#show" do
+    context  "with invalid api_key" do
+      it "returns a 401" do
+        get :show, id: user.id, api_key: 'blank'
+        expect(response.code).to eq '401'
+      end
+    end
+
+    context "with a valid api_key" do
+      render_views
+
+      let(:json) { JSON.parse(response.body) }
+
+      before do
+        # add links to user first
+        user.links << link1
```

Comment:
Another shortcut for this would be to edit lines 5 and 6:

`let(:link1) { create(:link, user_id: user.id) }`